libuuid/parse.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libuuid-tests
+ *
+ * Testing parse functions
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Damian Loewnau
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+#include <stdio.h>
+#include <uuid/uuid.h>
+
+#include <unity_fixture.h>
+
+
+#define TESTED_UUIDS_NUMBER 128
+#define ARRAY_SIZE          4
+
+const uuid_t raw[ARRAY_SIZE] = {
+	{ 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0 },
+	{ 0x0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0xa, 0xb, 0xc, 0xd, 0xe, 0xf },
+	{ 0x0, 0x10, 0x20, 0x30, 0x40, 0x50, 0x60, 0x70, 0x80, 0x90, 0xa0, 0xb0, 0xc0, 0xd0, 0xe0, 0xf0 },
+	{ 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff }
+};
+
+const char *correctStr[ARRAY_SIZE] = {
+	"00000000-0000-0000-0000-000000000000",
+	"00010203-0405-0607-0809-0a0b0c0d0e0f",
+	"00102030-4050-6070-8090-a0b0c0d0e0f0",
+	"ffffffff-ffff-ffff-ffff-ffffffffffff"
+};
```

Comment:
Nitpick - we can tie input-output vectors using array of structs instead of 2 separate arrays:
```
struct {
  uuid_t raw;
  const char *str;
} vector[ARRAY_SIZE] = {
  { { 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0 }, "00000000-0000-0000-0000-000000000000" },
....
```

Reply:
Thanks, I think next time I will use it